webclient/src/components/admin/Category.js

```diff
@@ -1,30 +1,31 @@
 import React, { Component, useEffect, useState } from 'react';
-import useFetch from 'hooks/useFetch';
-import useMutation from 'hooks/useMutation';
 import M from 'materialize-css';
 import { Link } from 'react-router-dom';
+import { useQuery, useMutation, queryCache } from 'react-query';
+
 import FileUpload from './utils/FileUpload';
+import { client } from '~/utils/client';
+import { QUERY_KEY_ADMIN_CATEGORY } from '~/constants/query-keys';
+import { API_ADMIN_CATEGORY } from '~/constants/api-url';
 
 const Category = props => {
-  const [
-    { response: categories, loading: fetchLoading, error: fetchError },
-    doFetch
-  ] = useFetch({ endpoint: '/api/admin/categories' });
-  const [{ error: mutationError }, upsertData] = useMutation();
-
-  const handleRemove = async slug => {
-    upsertData({
-      method: 'delete',
-      endpoint: `/api/admin/categories/${slug}`
+  const {
+    data: categories,
+    status: fetchStatus,
+    error: fetchError
+  } = useQuery(QUERY_KEY_ADMIN_CATEGORY, () => client(API_ADMIN_CATEGORY));
+
+  const [remove] = useMutation(handleRemove, {
+    onSuccess: () => queryCache.refetchQueries(QUERY_KEY_ADMIN_CATEGORY)
+  });
+
+  function handleRemove(slug) {
+    client(API_ADMIN_CATEGORY + `/${slug}`, {

```

Comment:
I think instead of something like this, it would be better to write [API_ADMIN_CATEGORY, { slug }]